EventEnabledInsurance/QuoteLifecycleSimulator/main.go

```diff
@@ -0,0 +1,217 @@
+// Customer data in order: Name, eMail, Address, USState, LicensePlate
+// Names generated using: https://www.name-generator.org.uk/quick/
+// Addresses generated using: https://www.randomlists.com/random-addresses
+// Licenes plates generated using: https://www.elfqrin.com/uscarlicenseplates.php
+var customers = [][]string {
+  {"Ronny Doyle", "RonnyDoyle@mail.com", "790 Arrowhead Court, Portsmouth", "VA", "WMC-9628"},
+  {"Nella Beard", "NBeard@mail.com", "8774 Inverness Dr., Janesville", "WI", "787-YWR"},
+  {"Andy Rosales", "AndyR@mail.com", "9783 Oxford St., Duluth", "GA", "GWL3149"},
+}
+const minAge = 21
+const maxAge = 80
+var damageDescriptions = []string{"Cracked windscreen", "Wheel fell off", "Dent in door", "Won't start"}
+const newClaimTicks = 3
+
+func main() {
+  psqlInfo := fmt.Sprintf("host=%s port=%d user=%s password=%s dbname=%s sslmode=disable",
+    host, port, user, password, dbname)
+  db, err := sql.Open("postgres", psqlInfo)
+  if err != nil {
+    panic(err)
+  }
+  defer db.Close()
+
+  err = db.Ping()
+  if err != nil {
+    panic(err)
+  }
+  log.Println("Successfully connected to postgres database")
+
+  count := countQuotes(db);
+  if(count<=0) {
+    log.Printf("No quotes in table, creating %d new mobile quotes", mobileTestRows)
+    for i := 0; i < mobileTestRows; i++ {
+      createNewClaim(db, true)
+    }
+  } else {
+    log.Printf("Found %d quotes in table", count)
+  }
+
+  ticksToNextNewClaim := newClaimTicks
+  for {
+    time.Sleep(time.Duration(tickMilliSeconds) * time.Millisecond)
+    processMobileClaim(db)
+    processNonMobileClaim(db)
+    ticksToNextNewClaim--
+    if(ticksToNextNewClaim <= 0) {
+      createNewClaim(db, false)
+      ticksToNextNewClaim = newClaimTicks
+    }
+  }
+}
+
+func processMobileClaim(db *sql.DB) () {
+  quoteID, claimStatus := getRandomClaim(db, true)
+  if(quoteID<0) {
+    log.Printf("No outstanding mobile claims found")
+  } else {
+    log.Printf("Found mobile claim with quoteID of %d and claimStatus of %d\n", quoteID, claimStatus)
+    claimStatus+=1
+    updateQuoteStatus(db, quoteID, claimStatus)
+  }
+}
+
+func processNonMobileClaim(db *sql.DB) () {
+  quoteID, claimStatus := getRandomClaim(db, false)
+  if(quoteID<0) {
+    log.Printf("No outstanding non-mobile claims found")
+  } else {
+    log.Printf("Found non-mobile claim with quoteID of %d and claimStatus of %d\n", quoteID, claimStatus)
+    claimStatus+=1
+    // For 2/3 of cases skip 3 and move straight to 4
+    if(claimStatus==3 && rand.Intn(3)>0) {
+      log.Printf("Skipping claimStatus straight to 4")
+      claimStatus=4
+    }
+    updateQuoteStatus(db, quoteID, claimStatus)
+  }
+}
+
+func createNewClaim(db *sql.DB, mobile bool) {
+  var source string
+  if(mobile) {
+    source = "Mobile"
+  } else {
+    source = pickRandomString(nonMobileSources)
+  }
+  var damageDescriptions = []string{"Cracked windscreen", "Wheel fell off", "Dent in door", "Won't start"}
```

Comment:
Seems like there are 2 declarations done for damageDescriptions?

Reply:
2nd definition removed, thanks!